tests/test_enterprise/views/test_grant_data_sharing_permissions.py

```diff
@@ -594,13 +596,10 @@ def test_post_course_specific_consent(
 
         assert resp.url.endswith(expected_redirect_url)  # pylint: disable=no-member
         assert resp.status_code == 302
-        if not defer_creation:
-            dsc.refresh_from_db()
-            assert dsc.granted is consent_provided
 
-        # we'll only create an enrollment record if (1) creation is not deferred, (2) consent has not been provided,
+        # we'll only create an enrollment record if (1) creation is not deferred, (2) consent has been provided,
         # (3) we provide a license_uuid, and (4) we provide a course _run_ id
-        if not defer_creation and not consent_provided and license_uuid and course_id.endswith('Demo_Course'):
+        if not defer_creation and consent_provided and license_uuid and course_id.endswith('Demo_Course'):
```

Comment:
Nice, this now matches https://github.com/edx/edx-enterprise/blob/31183667f211582ffe872740c5bde141b40e720a/enterprise/views.py#L975